src/jquery.interactive-world-map.js

```diff
@@ -39,104 +85,95 @@ $(function(){
             }
 
             if($data.colorhover){
-                $.fn.interactiveWorldMap.defaults.hover       = $data.colorhover
+                $.fn[pluginName].defaults.hover       = $data.colorhover
             }
             if($data.colorhighlighted){
-                $.fn.interactiveWorldMap.defaults.highlighted = $data.colorhighlighted;
+                $.fn[pluginName].defaults.highlighted = $data.colorhighlighted;
             }
         }
+    }
```

Comment:
The whole `setOptions` method seems very convoluted. Have a think about single responsibility - how can you restructure the method to make it more clearer. My suggestion would be to split this method into several other methods that all care about one thing, i.e. extracting `data-*` option attributes only, extracting `data-countries` only, etc.

Take a look at the `$.extend` function (https://api.jquery.com/jquery.extend/), it gives you really nice hash merging, like:
```
a = { x: 1 }
b = { y: 2 }
$.extend(a, b) --> { x: 1, y: 2 }
```